app/src/hog.c

```diff
@@ -85,6 +91,31 @@ static ssize_t read_hids_input_report(struct bt_conn *conn, const struct bt_gatt
                              sizeof(struct zmk_hid_keyboard_report_body));
 }
 
+static ssize_t write_hids_output_report(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+                                        const void *buf, uint16_t len, uint16_t offset,
+                                        uint8_t flags) {
+    if (flags & BT_GATT_WRITE_FLAG_PREPARE) {
+        return 0;
+    }
+
+    if (len != sizeof(struct zmk_hid_led_report_body)) {
+        LOG_ERR("LED report is malformed: length=%d", len);
+        return BT_GATT_ERR(BT_ATT_ERR_INVALID_ATTRIBUTE_LEN);
+    }
+
+    struct zmk_hid_led_report_body led_report;
+    memcpy(&led_report, (uint8_t *)buf + offset, sizeof(struct zmk_hid_led_report_body));
+
+    struct zmk_host_report_source source = {
+        .endpoint = ZMK_ENDPOINT_BLE,
+        .profile = bt_conn_index(conn),
```

Comment:
The connection index isn't the same as the profile #. Why is this info needed? Do we switch the LED indicator based on the active profile/connection? (If not, we should)

Reply:
Yes, we switch the LED indicators only on the active profile but we still receive reports from other profiles and we should handle them.